### PR TITLE
Support reversing bool sequence.

### PR DESCRIPTION
**Pull request by betterenvi**
This PR tries to fix issue [#13717](https://github.com/tensorflow/tensorflow/issues/13717).

### CODE REVIEW

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

File: tensorflow/core/kernels/reverse_sequence_op.cc [hunk 1]
```diff
@@ -174,7 +174,7 @@ class ReverseSequenceOp : public OpKernel {
   REGISTER_REVERSE_SEQUENCE(type, int32);      \
   REGISTER_REVERSE_SEQUENCE(type, int64);
 
-TF_CALL_NUMBER_TYPES(REGISTER_REVERSE_SEQUENCE_LEN);
+TF_CALL_POD_TYPES(REGISTER_REVERSE_SEQUENCE_LEN);
```
Comment (ebrevdo) [inline tensorflow/core/kernels/reverse_sequence_op.cc]: just add
```
TF_CALL_bool(REGISTER_REVERSE_SEQUENCE_LEN)
```

here
etc.

there are other GPU pod types that you're not including in your declaration in register_types; i recommend reverting that change.

File: tensorflow/python/ops/array_grad.py [hunk 2]
```diff
@@ -568,6 +568,8 @@ def _PadGrad(op, grad):
 # ReverseSequence is just a permutation.  The gradient permutes back.
 @ops.RegisterGradient("ReverseSequence")
 def _ReverseSequenceGrad(op, grad):
+  if op.inputs[0].dtype.base_dtype == ops.dtypes.bool:
```
Comment (ebrevdo) [inline tensorflow/python/ops/array_grad.py]: if op.get_attr("T") == dtypes.bool:
  ...

Comment (ebrevdo) [reply tensorflow/python/ops/array_grad.py]: (though you don't really need to add this special case for the grad at al)

**Review by ebrevdo — CHANGES_REQUESTED** [review]

**Comment by betterenvi** [comment]
@ebrevdo 
I had made some changes according to your comments.

**Review by ebrevdo — APPROVED** [review]
Thanks!

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Pull request merged**